components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2387,11 +2437,11 @@ private Group addDomainToUserMembers(Group group, String userStoreDomain) throws
      * Paginate a list of users names according to a given offset and a count.
      *
      * @param users  A list of unpaginated users.
-     * @param limit  The total number of results required (Zero will return all the users).
+     * @param limit  The total number of results required (ZERO will return all the users).
      * @param offset The starting index of the count (limit).
      * @return A list of paginated users
      */
-    private String[] paginateUsers(String[] users, int limit, int offset) {
+    private String[] paginateUsers(String[] users, Integer limit, int offset) {
```

Comment:
Is it necessary to change the method signature. I don't find a reason here

Reply:
fixed: 8f8c20292c4fac76778a8aa95490675023d7d063